grammar.js

```diff
@@ -223,6 +229,31 @@ module.exports = grammar({
             )
         },
 
+        _interval_date_spec: $ => {
+            const ci = s => new RegExp(caseInsensitive(s))
+
+            const from = seq(
+                choice(ci('since'), ci('from')),
+                $.whitespace,
+                $.date_spec,
+            );
+            const until = seq(
+                choice(ci('until'), ci('to')),
+                $.whitespace,
+                $.date_spec,
+            );
+
+            return choice(
+                from,
+                until,
+                seq(from, $.whitespace, until),
+                seq(ci('in'), $.whitespace, $.date_spec),
+
+                // just "next week", "last month", etc
+                alias($._relative_date_spec, $.date_spec),
```

Comment:
The goal is to support all of the following, for example:
- from last week
- to next month
- from last week to next month
- in last month
- last month

And these can all be combined with a (previously supported) period like daily/weekly, to be something like "daily in last month" or "weekly from last week to next month". 😵

Reply:
Very fancy!